[DOCS] Cookbook for customizing refund process (#12686)
Sylius/Sylius: docs/cookbook/refund/refund-process.rst

> How to customize refund process?

Reviewer: ```suggestion
How to customize the refund process?
```

================================

> The Refund Plugin provides possibility to refund full orders as well as partially.

Reviewer: ```suggestion
The Refund Plugin provides a possibility to refund orders fully as well as partially.
```

With this plugin there comes a process where credit memos and refund payments are created.

> In vanilla Plugin we are creating refund payment right after when the credit memo has been created.

Reviewer: ```suggestion
In the vanilla plugin, we are creating the refund payment right after the credit memo has been created.
```

> But what about situations where requirements ask as to change their places?

Reviewer: ```suggestion
But what about situations where requirements say to change this order?
```

We will find out about it in this tutorial.

> Changing order of refund process.

Reviewer: ```suggestion
Changing the order of the refund process
```

---------------------------------

> Let's say that you need to change the order of refund process.

Reviewer: ```suggestion
Let's say that you need to change the order of the refund process.
```

> There is an easy way to customize it. You just need to override the priority in service declaration in config file:

Reviewer: ```suggestion
There is an easy way to customize it. You just need to override the priority in service declaration in the config file:
```

.. code-block:: yaml

    # config/services.yaml

>     Sylius\RefundPlugin\ProcessManager\CreditMemoProcessManager:
>             tags:
>                 - { name: sylius_refund.units_refunded.process_step, priority: 0 }

Reviewer 2: ```suggestion
    Sylius\RefundPlugin\ProcessManager\CreditMemoProcessManager:
        tags:
            - { name: sylius_refund.units_refunded.process_step, priority: 0 }
```

Or like this:

.. code-block:: yaml

    # config/services.yaml

>     Sylius\RefundPlugin\ProcessManager\RefundPaymentProcessManager:
>             tags:
>                 - { name: sylius_refund.units_refunded.process_step, priority: 200 }

Reviewer 2: ```suggestion
    Sylius\RefundPlugin\ProcessManager\RefundPaymentProcessManager:
        tags:
            - { name: sylius_refund.units_refunded.process_step, priority: 200 }
```

The process will work according to the priority (descending).

.. tip::

>     You can find the default values of refund process in `%kernel.project_dir%/vendor/sylius/refund-plugin/src/Resources/config/services/event_bus.xml`

Reviewer: ```suggestion
    You can find the default config of the refund process in `%kernel.project_dir%/vendor/sylius/refund-plugin/src/Resources/config/services/event_bus.xml`
```

    on services tagged as `sylius_refund.units_refunded.process_step`

> After one of this changes, the refund process will be shifted and the Credit Memo will be generated after Refund Payment.

Reviewer: ```suggestion
After one of these changes, the refund process will be shifted and the Credit Memo will be generated after the Refund Payment.
```
